core/client/fs/src/main/java/alluxio/client/file/cache/CacheManagerWithShadowCache.java

```diff
@@ -0,0 +1,234 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.client.file.cache;
+
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+import alluxio.client.quota.CacheQuota;
+import alluxio.client.quota.CacheScope;
+import alluxio.metrics.MetricKey;
+import alluxio.metrics.MetricsSystem;
+
+import com.codahale.metrics.Counter;
+import com.google.common.hash.BloomFilter;
+import com.google.common.hash.Funnel;
+import com.google.common.hash.PrimitiveSink;
+
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.atomic.AtomicIntegerArray;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.concurrent.atomic.AtomicLongArray;
+import java.util.concurrent.atomic.AtomicReferenceArray;
+
+/**
+ * A wrapper class of CacheManager with shadow cache.
+ */
+public class CacheManagerWithShadowCache implements CacheManager {
+  private static final int NUM_BLOOM_FILTER = 4;
+  private static final int NUM_BLOOM_FILTER_EXPECT_INSERTION = 20_000_000;
+  private static final double NUM_BLOOM_FILTER_FPP = .00819;
+  private final CacheManager mCacheManager;
+  private final AtomicReferenceArray<BloomFilter<PageId>> mBFs =
+      new AtomicReferenceArray<BloomFilter<PageId>>(new BloomFilter[NUM_BLOOM_FILTER]);
+  private final AtomicIntegerArray mObjInserted = new AtomicIntegerArray(new int[NUM_BLOOM_FILTER]);
+  private final AtomicLongArray mByteInserted = new AtomicLongArray(new long[NUM_BLOOM_FILTER]);
+  ScheduledExecutorService mScheduler = Executors.newScheduledThreadPool(0);
+  // working set size window, in unit of second
+  private int mWindow = 20;
+  private int mCurrentFilterIndex = 0;
+  private int mShadowCachePages = 0;
+  private long mShadowCacheBytes = 0;
+  private AtomicLong mShadowCachePageRead = new AtomicLong(0);
+  private AtomicLong mShadowCachePageHit = new AtomicLong(0);
+  private AtomicLong mShadowCacheByteRead = new AtomicLong(0);
+  private AtomicLong mShadowCacheByteHit = new AtomicLong(0);
+
+  /**
+   * @param cacheManager the real cache manager
+   * @param window the past window in which is the working set
+   */
+  public CacheManagerWithShadowCache(CacheManager cacheManager, int window) {
```

Comment:
how about pass `conf` to here, as in the future, we may need more properties.
from conf, we can get them. It is easier to isolate change
```suggestion
  public CacheManagerWithShadowCache(CacheManager cacheManager, AlluxioConfiguration  conf) {
     window = conf.get ....
```